pandas/io/parsers.py

```diff
@@ -1662,14 +1662,22 @@ def __init__(self, src, **kwds):
             # GH 14671
             if (self.usecols_dtype == 'string' and
                     not set(usecols).issubset(self.orig_names)):
-                raise ValueError("Usecols do not match names.")
+                missing = [c for c in usecols if c not in self.orig_names]
+                raise ValueError(
+                    "Usecols do not match columns, "
+                    "columns expected but not found: %s" % missing
```

Comment:
https://github.com/pandas-dev/pandas/issues/16130 (it's being worked on in chunks; feel free to pitch in)